### ISSUE TITLE
Incorrect Gradients at Boundary Points for `torch.nn.functional.hardswish`

### ISSUE THREAD

**Issue by vwrewsge** [pytorch/pytorch#147801]
### 🐛 Describe the bug

The gradients of the hardswish function at boundary points (specifically at -3.0 and 3.0) are incorrect. The gradient at -3.0 should be 0, and the gradient at 3.0 should be 1.0. However, the current implementation produces incorrect values at these points.

# Code
```
import torch

# Test case to check hardswish_backward gradients at boundaries
input_values = torch.tensor([-3.0, 3.0, -2.0, 2.0], requires_grad=True)
out = torch.nn.functional.hardswish(input_values)
out.backward(torch.ones_like(input_values))

# Expected gradients:
# -3.0: should be 0 (flat region), but current code gives (x/3 +0.5) = -0.5
# 3.0: should be 1.0 (linear region), but current code gives 1.5
# -2.0: correct gradient (2*(-2)+3)/6 = -1/6 ≈ -0.1667
# 2.0: correct gradient (2*2+3)/6 = 7/6 ≈ 1.1667
expected_grad = torch.tensor([0.0, 1.0, -1/6, 7/6], dtype=torch.float32)

print(input_values.grad)
print(expected_grad)
```

# Output
```
tensor([-0.5000,  1.5000, -0.1667,  1.1667])
tensor([ 0.0000,  1.0000, -0.1667,  1.1667])
```

### Versions

PyTorch version: 2.6.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4

cc @ezyang @albanD @gqchen @pearu @nikitaved @soulitzer @Varal7 @xmfan

**Comment by soulitzer** [comment]
The gradients today are indeed not the min-norm ones. We'd review a PR to fix this.
